web/files/referencedby.go

```diff
@@ -10,6 +11,53 @@ import (
 	"github.com/labstack/echo/v4"
 )
 
+// ContentType is the official mime-type for JSON-API
+const ContentType = "application/vnd.api+json"
+
+// UpdatedReferences is a container for the updated list of referenced_by object
+// for an io.cozy.files document.
+type UpdatedReferences struct {
+	DocRev       string                 `json:"rev"`
+	ReferencedBy []couchdb.DocReference `json:"referencedBy"`
+}
+
+func NewUpdatedReferences(rev string, refs []couchdb.DocReference) UpdatedReferences {
+	u := UpdatedReferences{DocRev: rev}
+	u.ReferencedBy = make([]couchdb.DocReference, len(refs))
+	copy(u.ReferencedBy, refs)
```

Comment:
Why do you make a copy here?

Reply:
Well, first, I had issues with pointer de-renferences when I first assigned the slice directly to the struct field.
And then I thought it would be a good idea to copy those values instead of having a reference to the original slice. But I might be wrong about slices being passed as references.